src/snappy.rs

```diff
@@ -246,8 +246,38 @@ where
         }
     }
 
-    fn poll_flush(self: Pin<&mut Self>, _cx: &mut Context) -> Poll<Result<()>> {
-        Poll::Ready(Ok(()))
+    fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<()>> {
+        let this = &mut *self;
+
+        while this.output_start != this.output_end {
```

Comment:
It looks like in every case of `match` this loop short circuits.

Reply:
Could this be switched to an `if`, and then instead of `break` just let it fall-through. I like to satisfy `clippy` if possible. Besides that looks great! I can get a new release out ASAP.

Reply:
I figured this part is never triggered, so I removed it